### PR TITLE
Implement socket communication with LeCroy

### PR DESCRIPTION
**Pull request by wsxrdv**

### CODE REVIEW

File: scaaml/capture/scope/lecroy/lecroy_communication.py [hunk 1]
```diff
@@ -185,3 +189,157 @@ def query_binary_values(self,
...
+    @make_custom_exception
+    def get_waveform(self, channel: LECROY_CHANNEL_NAME_T) -> LecroyWaveform:
+        """Get a LecroyWaveform object representing a single waveform.
+
+        Args:
+          channel (LECROY_CHANNEL_NAME_T): The name of queried channel.
+        """
+        raw_data = self.query_binary_values(f"{channel}:WAVEFORM?")
+        assert raw_data[:6] == b"ALL,#9"  # followed by 9 digits for size
+        raw_data = raw_data[15:-1]  # last is linefeed
```
[the remark this answers is not in the document]
Comment (kralka) [reply scaaml/capture/scope/lecroy/lecroy_communication.py]: I guess we can suppose that the message does not end in more blank characters, right?

File: scaaml/capture/scope/lecroy/lecroy_communication.py [hunk 2]
```diff
@@ -185,3 +189,157 @@ def query_binary_values(self,
...
+                spare,  # unused
+                total_bytes) = unpack(self._lecroy_command_header, header)
+
+            # Delete unused values
+            del header_version
+            del sequence_number
+            del spare
+
+            # Buffer for the current portion of data
+            buffer = b""
```
[the remark this answers is not in the document]
Comment (kralka) [reply scaaml/capture/scope/lecroy/lecroy_communication.py]: Good point, I always mess up `bytes` and `bytearray`. Do we believe that `bytearray.extend` is amortized ok (similar to append)? Or do we want to be extra cautious not to timeout during communication and join at the end? Both should be ok, just asking: is this efficiency -- worst case complexity or cautious not to timeout?

[the remark this answers is not in the document]
Comment (jmichelp) [reply scaaml/capture/scope/lecroy/lecroy_communication.py]: Should be good as is.

File: scaaml/capture/scope/lecroy/lecroy_communication.py [hunk 3]
```diff
@@ -178,10 +184,165 @@ def query_binary_values(self,
...
+    @make_custom_exception
+    def get_waveform(self, channel: LECROY_CHANNEL_NAME_T) -> LecroyWaveform:
+        """Get a LecroyWaveform object representing a single waveform.
+
+        Args:
+          channel (LECROY_CHANNEL_NAME_T): The name of queried channel.
+        """
+        raw_data = self.query_binary_values(f"{channel}:WAVEFORM?")
+        assert raw_data[:6] == b"ALL,#9"  # followed by 9 digits for size
+        raw_data = raw_data[15:]
```
[the remark this answers is not in the document]
Comment (kralka) [reply scaaml/capture/scope/lecroy/lecroy_communication.py:258]: Done

**Review by jmichelp — APPROVED** [review]

**Pull request merged**